src/app/components/client/InputField.tsx

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, useRef } from "react";
+import { AriaTextFieldProps, useTextField } from "react-aria";
+import styles from "./InputField.module.scss";
+
+export type Props = {
+  children: ReactNode;
+};
+
+export const InputField = (props: AriaTextFieldProps) => {
+  const { errorMessage, isRequired, label, validationState, value } = props;
+  const inputRef = useRef(null);
+  const { errorMessageProps, inputProps, labelProps } = useTextField(
```

Comment:
Oh this is awesome, I was wondering how we were going to accessibly do custom validation, but I didn't realise `useTextField` existed - nice find!